src/ui/menu.c

```diff
@@ -498,7 +476,7 @@ void meta_window_menu_popup(MetaWindowMenu* menu, int root_x, int root_y, int bu
 	pt->x = root_x / scale;
 	pt->y = root_y / scale;
 
-	gtk_menu_popup(GTK_MENU (menu->menu), NULL, NULL, popup_position_func, pt, button, timestamp);
+	gtk_menu_popup_at_pointer (GTK_MENU (menu->menu), NULL);
```

Comment:
`pt` is now unused...

Reply:
Should i really do that?
```
diff --git a/src/ui/menu.c b/src/ui/menu.c
index 01f09c5..9bbf51d 100644
--- a/src/ui/menu.c
+++ b/src/ui/menu.c
@@ -467,14 +467,7 @@ meta_window_menu_new   (MetaFrames         *frames,
 
 void meta_window_menu_popup(MetaWindowMenu* menu, int root_x, int root_y, int button, guint32 timestamp)
 {
-	GdkPoint* pt = g_new(GdkPoint, 1);
-	gint scale;
-
-	g_object_set_data_full(G_OBJECT(menu->menu), "destroy-point", pt, g_free);
-
-	scale = gtk_widget_get_scale_factor (menu->menu);
-	pt->x = root_x / scale;
-	pt->y = root_y / scale;
+	g_object_set_data_full(G_OBJECT(menu->menu), "destroy-point", NULL, g_free);
 
 	gtk_menu_popup_at_pointer (GTK_MENU (menu->menu), NULL);
```
It seems to be working in runtime...

Reply:
`g_object_set_data_full` is also unneeded... It was used to free allocated `GdkPoint`.

Reply:
Hmm, i noticed that i already killed using `alt-spacebar` to open the window-menu with first commit.
Maybe because of using `current event` (NULL) with gtk_menu_popup_at_pointer ?

Reply:
Right, if there is no `GdkEvent` then how GTK/GDK can know where to popup menu?

https://developer.gnome.org/gtk3/stable/gtk3-General.html#gtk-get-current-event
You can pass `NULL` if you know that `gtk_get_current_event` will return event.

I don't remember details, but I think keybindings does not use gtk/gdk... In `metacity` I changed that popup function parameters so I can pass `GdkEvent`. There are places where you have it, in other places I just created new event from x event data.